src/App.jsx

```diff
@@ -1,9 +1,71 @@
-import './App.css';
+import "./App.css";
+import { useState, useEffect } from "react";
+import { Route, Routes, Link } from "react-router-dom";
+import ContactList from "./components/ContactList";
+import ContactProfile from "./components/ContactProfile";
+import CreateContact from "./components/CreateContact";
 
-function App() {
-    return (
-        <p>Hello, world!</p>
+export default function App() {
+  const [contacts, setContacts] = useState([]);
+
+  useEffect(() => {
+    console.log(
+      "Running my effect once, note: dependency array contains []..."
     );
-}
+    fetchContacts();
+  }, []);
+  function fetchContacts() {
+    fetch(`https://boolean-api-server.fly.dev/jacobchenjensen/contact`)
+      .then((response) => response.json())
+      .then((data) => {
+        setContacts(data);
+      })
+      .catch((error) => {
+        console.error("Error fetching contact list:", error);
+      });
+  }
 
-export default App;
+  return (
+    <>
+      <header>
+        <h1>Welcome to your Address Book!</h1>
+        <nav>
+          <ul>
+            <li>
+              <Link to="/contacts" onClick={() => fetchContacts()}>
+                View All Contacts
+              </Link>
+            </li>
+            <li>
+              <Link to="/create">Create New Contact</Link>
+            </li>
+          </ul>
+        </nav>
+      </header>
+
+      <Routes>
+        <Route
+          path="/contacts"
+          element={
+            <ContactList contacts={contacts} fetchContacts={fetchContacts} />
+          }
+        />
+        <Route
+          path="/create"
+          element={
+            <CreateContact
+              setContacts={setContacts}
+              fetchContacts={fetchContacts}
```

Comment:
why are you re-fetching? the POST to the create contact should return back all the right data so you can just setContacts and that;s it